CONTRIBUTING.rst

```diff
@@ -946,6 +946,23 @@ Most of our coding style rules are enforced programmatically by flake8 and pylin
 on every pull request), but there are some rules that are not yet automated and are more Airflow specific or
 semantic than style
 
+Using Asserts Outside Tests
+---------------------------
+
+Our community agreed that to various reasons we **do not use ``assert`` in production code** of Apache Airflow.
+For details check the relevant [mailing list thread](https://lists.apache.org/thread.html/bcf2d23fcd79e21b3aac9f32914e1bf656e05ffbcb8aa282af497a2d%40%3Cdev.airflow.apache.org%3E).
+
+In other words instead of doing:
+
+.. code-block:: python
```

Comment:
The code example seems cannot be rendered properly ? Possibly due to the indentation.

<img width="1081" alt="airflow_CONTRIBUTING_rst_at_09da71fee1778cf352741b0bcf7c4b7c4ce1fd30_·_apache_airflow" src="https://user-images.githubusercontent.com/11539188/115951977-f501cf00-a4e3-11eb-945a-3ee0cd9140ca.png">

Reply:
<img width="1001" alt="Screenshot 2021-04-24 at 10 06 49" src="https://user-images.githubusercontent.com/9528307/115952166-cdf7cd00-a4e4-11eb-9b54-b1596d520d8c.png">
 Looks good now, Thanks!

Reply:
I also notice this:

<img width="736" alt="Add_code_style_note__no_asserts_by_turbaszek_·_Pull_Request__15512_·_apache_airflow" src="https://user-images.githubusercontent.com/11539188/115952276-524a5000-a4e5-11eb-9397-7102ab5c258d.png">

It's quite nit and I don't have a clue how to make it more "beautiful", so will approve first.

Thanks @turbaszek 👍

Reply:
Should be fixed now, it seems that code snippet cannot be rendered within bold text so I removed the emphasis.